lib/fileutils.rb

```diff
@@ -8,6 +8,85 @@
 
 # Namespace for file utility methods for copying, moving, removing, etc.
 #
+# == What's Here
+#
+# First, what’s elsewhere. \Module \FileUtils:
+#
+# - Inherits from {class Object}[https://docs.ruby-lang.org/en/master/Object.html].
+# - Supplements {class File}[https://docs.ruby-lang.org/en/master/File.html]
+#   (but is not included or extended there).
+#
+# Here, module \FileUtils provides methods that are useful for:
+#
+# - {Creating}[rdoc-ref:FileUtils@Creating].
+# - {Deleting}[rdoc-ref:FileUtils@Deleting].
+# - {Querying}[rdoc-ref:FileUtils@Querying].
+# - {Setting}[rdoc-ref:FileUtils@Setting].
+# - {Comparing}[rdoc-ref:FileUtils@Comparing].
+# - {Copying}[rdoc-ref:FileUtils@Copying].
+# - {Moving}[rdoc-ref:FileUtils@Copying].
+#
+# === Creating
+#
+# - ::mkdir: Creates directories.
+# - ::mkdir_p, ::makedirs, ::mkpath: Creates directories,
+#   also creating ancestor directories as needed.
+# - ::link_entry: Creates a hard link.
+# - ::ln, ::link: Creates hard links.
+# - ::ln_s, ::symlink: Creates symbolic links.
+# - ::ln_sf: Creates symbolic links, overwriting if necessary.
+#
+# === Deleting
+#
+# - ::remove_dir: Removes a directory and its descendants.
+# - ::remove_entry: Removes an entry, including its descendants if it is a directory.
+# - ::remove_entry_secure: Like ::remove_entry, but removes securely.
+# - ::remove_file: Removes a file entry.
+# - ::rm, ::remove: Removes entries.
+# - ::rm_f, ::safe_unlink: Like ::rm, but removes forcibly.
+# - ::rm_r: Removes entries and their descendants.
+# - ::rm_rf, ::rmtree: Like ::rm_r, but removes forcibly.
+# - ::rmdir: Removes directories and their descendants.
```

Comment:
`rmdir` does not remove descendants (only empty directories)

```ruby
irb(main):001:0> FileUtils.mkdir("test")
=> ["test"]
irb(main):002:0> FileUtils.touch("test/foo")
=> ["test/foo"]
irb(main):003:0> FileUtils.rmdir("test")
/Users/peter/.rubies/ruby-3.1.2/lib/ruby/3.1.0/fileutils.rb:261:in `rmdir': Directory not empty @ dir_s_rmdir - test (Errno::ENOTEMPTY)
```

Reply:
Fixed.